src/endpoints/authentication/impl.cpp

```diff
@@ -20,120 +21,484 @@
-
-                CheckAuthCredentialsInput input{};
-                username.copy(input.username, sizeof(CheckAuthCredentialsInput::username));
-                zone.copy(input.zone, sizeof(CheckAuthCredentialsInput::zone));
-
-                namespace adm = irods::experimental::administration;
-                const adm::user_password_property prop{password, rodsadmin_password};
-                const auto obfuscated_password = irods::experimental::administration::obfuscate_password(prop);
-                obfuscated_password.copy(input.password, sizeof(CheckAuthCredentialsInput::password));
-
-                int* correct{};
-
-                auto conn = irods::get_connection(rodsadmin_username);
-
-                if (const auto ec = rc_check_auth_credentials(static_cast<RcComm*>(conn), &input, &correct); ec < 0) {
-                    log::error("{}: Error verifying native authentication credentials for user [{}]: error code [{}].", fn, username, ec);
-                }
-                else {
-                    log::debug("{}: correct = [{}]", fn, fmt::ptr(correct));
-                    log::debug("{}: *correct = [{}]", fn, (correct ? *correct : -1));
-                    login_successful = (correct && 1 == *correct);
-                }
-            }
-            catch (const irods::exception& e) {
-                log::error("{}: Error verifying native authentication credentials for user [{}]: {}", fn, username, e.client_display_what());
-            }
-            catch (const std::exception& e) {
-                log::error("{}: Error verifying native authentication credentials for user [{}]: {}", fn, username, e.what());
-            }
-
-            if (!login_successful) {
-                return _sess_ptr->send(fail(status_type::unauthorized));
-            }
-
-            static const auto seconds = irods::http::globals::configuration().at(nlohmann::json::json_pointer{"/http_server/authentication/basic/timeout_in_seconds"}).get<int>();
-            auto bearer_token = irods::process_stash::insert(authenticated_client_info{
-                .auth_scheme = authorization_scheme::basic,
-                .username = std::move(username),
-                .password = std::move(password),
-                .expires_at = std::chrono::steady_clock::now() + std::chrono::seconds{seconds}
-            });
-
-            response_type res{status_type::ok, _req.version()};
-            res.set(field_type::server, irods::http::version::server_name);
-            res.set(field_type::content_type, "text/plain");
-            res.keep_alive(_req.keep_alive());
-            res.body() = std::move(bearer_token);
-            res.prepare_payload();
-
-            return _sess_ptr->send(std::move(res));
-        });
-    } // authentication
-} // namespace irods::http::endpoint
+	auto hit_token_endpoint(std::string encoded_body) -> nlohmann::json
```

Comment:
The only two calls to this function pass an r-value to the arg, so it should be a move if I recall correctly.

We move the `encoded_body` later on, not sure if this would influence the decision on whether we capture by value in the case we receive an l-value?

Reply:
Ok. Feel free to mark resolved.